File: bronze/10811/c++/main.cpp
#include <iostream>

int main()
{
    int n, m, b, e, temp;
    std::cin >> n >> m;

    int array[n];
    for (int i = 0; i <= n; i++) {
        array[i] = i +1;
    }

    for (int _ = 0; _ < m; _++) {
        std::cin >> b >> e;
        for (int i = 0; i < ((e - b +1) / 2); i++) {
            temp = array[b + i -1];
            array[b + i -1] = array[e - i -1];
            array[e - i -1] = temp;
        }
    }

    for (int data : array) {
        std::cout << data << " ";
    }
    return 0;
}

File: bronze/10811/python/main.py

if (__name__ == "__main__"):
    n, m = map(int, input().split(' '))
    array = list()
    for i in range(n):
        array.append(i+1)
    for _ in range(m):
        b, e = map(int, input().split())
        for i in range(int((e - b +1) / 2)):
            num1 = b + i -1
            num2 = e - i -1
            temp = array[num1]
            array[num1] = array[num2]
            array[num2] = temp
    for data in array:
        print(data, end=' ')
